src/VirtualClient/VirtualClient.Actions/DCGMI/DCGMIDiagExecutor.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO.Abstractions;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Logging;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Platform;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// The DCGMI Diag Executor for GPU
+    /// </summary>
+    [UnixCompatible]
+    public class DCGMIDiagExecutor : VirtualClientComponent
+    {
+        private ISystemManagement systemManagement;
+        private IStateManager stateManager;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DCGMIDiagExecutor"/> class.
+        /// </summary>
+        public DCGMIDiagExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+            : base(dependencies, parameters)
+        {
+            this.systemManagement = this.Dependencies.GetService<ISystemManagement>();
+            this.stateManager = this.systemManagement.StateManager;
+        }
+
+        /// <summary>
+        /// Level specifies which set of tests we need to run.
+        /// </summary>
+        public string Level
+        {
+            get
+            {
+                this.Parameters.TryGetValue(nameof(DCGMIDiagExecutor.Level), out IConvertible level);
+                return level?.ToString();
+            }
+        }
+
+        /// <summary>
+        /// Initializes enviroment to run DCGMI Diag.
+        /// </summary>
+        /// <param name="telemetryContext"></param>
```

Comment:
I know it's repetitive work, but would be nice if you could fill the XML comments.